### PR TITLE
Plugins: Add upload selectors

### PR DESCRIPTION
**Pull request by seear**
Add selectors with unit tests for plugin upload state. Similar to the existing [theme upload selectors](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/upload-theme/selectors.js).

Part of #15665. No visual differences.

**To Test**
* Run unit tests

### CODE REVIEW

File: client/state/selectors/is-plugin-upload-complete.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Indicates whether a plugin upload has completed
+ * for the given site.
+ *
+ * @param {Object} state Global state tree
+ * @param {number} siteId the site ID
+ * @return {boolean} true if plugin upload is complete
+ */
+export default function isPluginUploadComplete( state, siteId ) {
+	return !! (
+		( ! isPluginUploadInProgress( state, siteId ) ) &&
```
Comment (sirreal) [inline client/state/selectors/is-plugin-upload-complete.js]: Minor: Parens aren't necessary, leave if you think it clarifies intent.

File: client/state/selectors/test/get-plugin-upload-error.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const siteId = 77203074;
+const error = {
+	error: 'folder_exists',
+	message: 'folder_exists',
+};
+
+describe( 'getPluginUploadError', () => {
+	it( 'should return null by default', () => {
+		const state = deepFreeze( {
```
Comment (tyxla) [inline client/state/selectors/test/get-plugin-upload-error.js]: What's the benefit of `deepFreeze`-ing state for testing selectors? Makes sense to verify prevention of state mutation when testing reducers, but for selectors?

File: client/state/selectors/test/get-plugin-upload-error.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			plugins: {
+				upload: {
+					uploadError: {}
+				}
+			}
+		} );
+		expect( getPluginUploadError( state, siteId ) ).to.be.null;
+	} );
+
+	it( 'should return current value for site', () => {
```
Comment (tyxla) [inline client/state/selectors/test/get-plugin-upload-error.js]: Is it worth to add a test case where state has an error for another site, to verify that the selector is not incorrectly returning an error for another site?

This question applies to the rest of the tests.

File: client/state/selectors/get-plugin-upload-progress.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Returns percentage of plugin zip uploaded to a site.
+ *
+ * @param {Object} state Global state tree
+ * @param {number} siteId the site ID
+ * @return {number} % of file uploaded so far
+ */
+export default function getPluginUploadProgress( state, siteId ) {
+	return get( state.plugins.upload.progressPercent, siteId, 0 );
```
Comment (seear) [inline client/state/selectors/get-plugin-upload-progress.js]: > Or we could just use isPluginUploadInProgress for that?

Correct.

File: client/state/selectors/test/get-plugin-upload-error.js (hunk 3 above)
Comment (seear) [inline client/state/selectors/test/get-plugin-upload-error.js]: Yeah I used to do those, but thinking now that `keyedReducer` makes those tests unnecessary? Unless `keyedReducer` breaks, but that is a different concern.

My general view on tests for actions/reducers/selectors is that we have broken these things down into such small parts that most of the value of the tests are as a part of the development process, so it is a judgement call as to how far to go.

Unless we are talking about something like a complex selector, where unit tests can be useful as documentation and guarding against regressions.

File: client/state/selectors/test/get-plugin-upload-error.js (hunk 2 above)
Comment (seear) [inline client/state/selectors/test/get-plugin-upload-error.js]: No benefit I think, and I'm going to remove it 👍

**Comment by seear** [comment]
Regarding the simultaneous upload question, I think there are two ways we can go with it. 1) persist all the state (the approach activity-log rewind takes) which would effectively prevent simultaneous uploads. 2) index all state by an upload ID.

Since this state will most likely be shared with atom initiations, I think I favour option 1, since simultaneous AT initiations are not desirable.

**Pull request merged**